test/Microsoft.ML.Functional.Tests/IntrospectiveTraining.cs

```diff
@@ -0,0 +1,431 @@
+                hasHeader: TestDatasets.iris.fileHasHeader,
+                separatorChar: TestDatasets.iris.fileSeparator);
+
+            // Compose the transformation.
+            var pipeline = mlContext.Transforms.Concatenate("Features", Iris.Features)
+                .Append(mlContext.Transforms.Normalize("Features", mode: NormalizingEstimator.NormalizerMode.MinMax));
+
+            // Fit the pipeline.
+            var model = pipeline.Fit(data);
+
+            // Extract the normalizer from the trained pipeline.
+            // TODO #2854: Extract the normalizer parameters.
+            var normalizer = model.LastTransformer;
+        }
+
+        /// <summary>
+        /// I can take an existing model file and inspect what transformers were included in the pipeline. 	 
+        /// </summary>
+        [Fact]
+        public void InspectPipelineContents()
+        {
+            var mlContext = new MLContext(seed: 1, conc: 1);
+
+            // Get the dataset.
+            var data = mlContext.Data.LoadFromTextFile<HousingRegression>(GetDataPath(TestDatasets.housing.trainFilename), hasHeader: true);
+
+            // Create a pipeline to train on the housing data.
+            var pipeline = mlContext.Transforms.Concatenate("Features", HousingRegression.Features)
+                .Append(mlContext.Regression.Trainers.FastForest(numLeaves: 5, numTrees: 3));
+
+            // Fit the pipeline.
+            var model = pipeline.Fit(data);
+
+            // Inspect the transforms in the trained pipeline.
+            var expectedTypes = new Type[] {typeof(ColumnConcatenatingTransformer),
+                typeof(RegressionPredictionTransformer<FastForestRegressionModelParameters>)};
+            var expectedColumns = new string[][] {
+                new string[] { "Features" },
+                new string[] { "Score" },
+            };
+            int i = 0;
+            var currentSchema = data.Schema;
+            foreach (var transformer in model)
+            {
+                // It is possible to get the type at runtime.
+                Assert.IsType(expectedTypes[i], transformer);
+                
+                // It's also possible to inspect the schema output from the transform.
+                currentSchema = transformer.GetOutputSchema(currentSchema);
+                foreach (var expectedColumn in expectedColumns[i])
+                {
+                    var column = currentSchema.GetColumnOrNull(expectedColumn);
+                    Assert.NotNull(column);
+                }
+                // And we can see that future columns do not yet exist.
+                if (i < expectedColumns.Length - 1)
+                    foreach (var expectedColumn in expectedColumns[i+1])
+                    {
+                        var column = currentSchema.GetColumnOrNull(expectedColumn);
+                        Assert.Null(column);
+                    }
+                i++;
```

Comment:
Seems a bit complex and overkill. We only have two transforms in the chain, so this will run for the first transform and will check that the outputschema does not contain `Score`.